stdlib/public/runtime/Array.cpp

```diff
@@ -0,0 +1,212 @@
+//===------------- Array.cpp - Swift Array Operations Support -------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// Implementations of the array runtime functions.
+//
+// arrayInitWithCopy(T *dest, T *src, size_t count, M* self)
+// arrayInitWithTake(NoAlias|FrontToBack|BackToFront)(T *dest, T *src,
+//                                                    size_t count, M* self)
+// arrayAssignWithCopy(NoAlias|FrontToBack|BackToFront)(T *dest, T *src,
+//                                                      size_t count, M* self)
+// arrayAssignWithTake(T *dest, T *src, size_t count, M* self)
+// arrayDestroy(T *dst, size_t count, M* self)
+//
+//===----------------------------------------------------------------------===//
+
+#include "swift/Runtime/Config.h"
+#include "swift/Runtime/HeapObject.h"
+#include "swift/Runtime/Metadata.h"
+
+using namespace swift;
+
+enum class ArrayCopy : unsigned {
```

Comment:
Nitpick: please put these in an anonymous namespace, so they don't cause ODR problems (unlikely but I think possible).

Reply:
Good point.